posthog/temporal/data_imports/pipelines/mongo/mongo.py

```diff
@@ -14,61 +13,133 @@
+
 from posthog.temporal.data_imports.pipelines.pipeline.typings import SourceResponse
+from posthog.temporal.data_imports.pipelines.source import config
 from posthog.temporal.data_imports.pipelines.pipeline.utils import (
     DEFAULT_PARTITION_TARGET_SIZE_IN_BYTES,
-    table_from_iterator,
 )
-from posthog.temporal.data_imports.pipelines.source import config
-from posthog.temporal.data_imports.pipelines.source.sql import Column, Table
-from posthog.temporal.data_imports.pipelines.pipeline.consts import DEFAULT_CHUNK_SIZE
-from posthog.warehouse.models.ssh_tunnel import SSHTunnel, SSHTunnelConfig
 from posthog.warehouse.types import IncrementalFieldType, PartitionSettings
 
 
 @config.config
 class MongoSourceConfig(config.Config):
     connection_string: str
-    ssh_tunnel: SSHTunnelConfig | None = None
-
-    @classmethod
-    def from_dict(cls, data: dict[str, Any]) -> MongoSourceConfig:
-        """Create MongoSourceConfig from dictionary, handling SSH tunnel configuration."""
-        ssh_tunnel_config = None
-
-        # Handle SSH tunnel configuration from job_inputs format
-        if data.get("ssh_tunnel_enabled"):
-            ssh_tunnel_config = SSHTunnelConfig(
-                enabled=data.get("ssh_tunnel_enabled", False),
-                host=data.get("ssh_tunnel_host"),
-                port=data.get("ssh_tunnel_port"),
-                auth_type=data.get("ssh_tunnel_auth_type"),
-                username=data.get("ssh_tunnel_auth_type_username"),
-                password=data.get("ssh_tunnel_auth_type_password"),
-                passphrase=data.get("ssh_tunnel_auth_type_passphrase"),
-                private_key=data.get("ssh_tunnel_auth_type_private_key"),
-            )
-        # Handle SSH tunnel configuration from nested format
-        elif "ssh_tunnel" in data and isinstance(data["ssh_tunnel"], dict):
-            ssh_tunnel_data = data["ssh_tunnel"]
-            if ssh_tunnel_data.get("enabled"):
-                auth_data = ssh_tunnel_data.get("auth", {})
-                ssh_tunnel_config = SSHTunnelConfig(
-                    enabled=ssh_tunnel_data.get("enabled", False),
-                    host=ssh_tunnel_data.get("host"),
-                    port=ssh_tunnel_data.get("port"),
-                    auth_type=auth_data.get("type"),
-                    username=auth_data.get("username"),
-                    password=auth_data.get("password"),
-                    passphrase=auth_data.get("passphrase"),
-                    private_key=auth_data.get("private_key"),
-                )
-
-        return cls(
-            connection_string=data["connection_string"],
-            ssh_tunnel=ssh_tunnel_config,
+
+
+def _process_nested_value(value: Any) -> Any:
+    """Process a nested value, converting ObjectIds to strings."""
+    if isinstance(value, ObjectId):
+        return str(value)
+    elif isinstance(value, dict):
+        return _process_nested_object(value)
```

Comment:
if you did `return { key: _process_nested_value(value) for key, value in obj.items() }` in here, then you wouldn't need the `_process_nested_object` function